core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/PropertiesUtils.java

```diff
@@ -421,6 +421,10 @@ public static Object getTypedPropertyFromUri(AbstractCamelModelElement selectedE
 		if (CamelComponentUtils.isTextProperty(p)) {
 			return val;
 		}
+		
+		if (CamelComponentUtils.isClassProperty(p) && !p.getKind().equals(AbstractCamelModelElement.NODE_KIND_ELEMENT) ) {
```

Comment:
given that the whole handling for this will change with next version I wouldn't invest into doing that now.